core/include/scipp/core/element/arithmetic.h

```diff
@@ -47,10 +60,21 @@ template <class... Ts> struct divide_types_t {
   using types = arithmetic_type_pairs;
 };
 
-constexpr auto plus =
-    overloaded{add_types_t{}, [](const auto a, const auto b) { return a + b; }};
-constexpr auto minus =
-    overloaded{add_types_t{}, [](const auto a, const auto b) { return a - b; }};
+constexpr auto plus = overloaded{
+    add_types_t{}, [](const auto a, const auto b) { return a + b; },
+    [](const scipp::core::time_point &a, const scipp::core::time_point &b) {
+      // a + b for time_point is undefined. Can't throw from here
+      // so return false
+      (void)a;
+      (void)b;
+      return false;
+    }};
+constexpr auto minus = overloaded{
+    add_types_t{}, [](const auto a, const auto b) { return a - b; },
+    [](const scipp::core::time_point &a, const scipp::core::time_point &b) {
+      return std::chrono::duration_cast<std::chrono::nanoseconds>(a - b)
+          .count();
+    }};
```

Comment:
This highlights an issue we are having with what we have done so far: `scipp::core::time_point` hard-codes the duration (resolution) in the data type. However, we cannot prevent setting a different unit in scipp, so there may be a mismatch. Not sure what can be done to fix this, I will need to think about it.

Reply:
We may considering using a custom `int` type instead to create our own time_point, without the hard-coded unit.

Reply:
so it's been replaced with custom `time_point`